parso/python/errors.py

```diff
@@ -730,9 +742,34 @@ def is_issue(self, node):
 @ErrorFinder.register_rule(type='star_expr')
 class _StarExprRule(SyntaxRule):
     message_iterable_unpacking = "iterable unpacking cannot be used in comprehension"
-    message_assignment = "can use starred expression only as assignment target"
 
     def is_issue(self, node):
+        def check_delete_starred(node):
+            while node.parent is not None:
+                node = node.parent
+                if node.type == 'del_stmt':
+                    return True
+                if node.type not in (*_STAR_EXPR_PARENTS, 'atom'):
+                    return False
+            return False
```

Comment:
You don't have to re-implement parent searching system here, I think you reduce this function (and your usage to) just this;
```
nearest_ancestor = search_ancestor(node, 'del_stmt', 'atom', *_STAR_EXPR_PARENTS)
if nearest_ancestor == 'del_stmt':
    ...
```

Reply:
`search_ancestor` locates the nearest ancestor of one of the given types, but here I need to find the nearest ancestor which is **NOT** of the given types (`(*_STAR_EXPR_PARENTS, 'atom')`).

Reply:
Ah, I see.